src/handler.ts

```diff
@@ -2,48 +2,64 @@ import { DynamoDBStreamEvent } from 'aws-lambda';
 import { DynamoDB } from 'aws-sdk';
 import { createTimestampRecord } from './createTimestampRecord';
 import { LegacyVehicleRecord } from './interfaces/LegacyVehicleRecord';
+import { BatchItemFailuresResponse } from './interfaces/BatchItemFailure';
 import logger from './util/logger';
 import config from './config';
 
 const client = new DynamoDB.DocumentClient();
 
-const handler = async (event: DynamoDBStreamEvent): Promise<void> => {
+const handler = async (event: DynamoDBStreamEvent): Promise<BatchItemFailuresResponse> => {
+  const dynamoDbPromises = [];
+
   const sendResponse = {
     SuccessCount: 0,
     FailCount: 0,
   };
 
+  const res: BatchItemFailuresResponse = {
+    batchItemFailures: [],
+  };
+
   logger.info('Received event:', JSON.stringify(event, null, 2));
+  logger.info(`Received ${event.Records.length} records from technical-records DynamoDB`);
+
   for (const dbRecord of event.Records) {        
     if (dbRecord.eventName === 'REMOVE') {
-      logger.info('remove event - ignoring');
+      logger.info('REMOVE event - ignoring');
       continue;
     }
 
     const newImage = DynamoDB.Converter.unmarshall(dbRecord.dynamodb.NewImage) as LegacyVehicleRecord;
-
     const techRecords = newImage.techRecord;
 
     delete newImage.techRecord;
 
-    logger.info(`processing ${techRecords.length} records`);
-    
-    for (const record of techRecords) {
-      const newRecord = createTimestampRecord(newImage, record);
-      try {
-        await (client.put({
-          TableName: config.dynamoDb.target,
-          Item: newRecord,
-        })).promise();
+    logger.info(`Processing ${techRecords.length} tech records for vehicle with systemNumber: ${newImage.systemNumber} and vin: ${newImage.vin}`);
+
+    dynamoDbPromises.push(...techRecords.map((techRecord) => {
```

Comment:
```suggestion
    const dynamoDbPromises = techRecords.map((techRecord) => {
```

I think you'd be able to do this, don't see why you'd have to create then empty array to then push to it.

Reply:
So it's because we're iterating over a batch of records, but each record can produce multiple promises. It's possible to do it with reduce but it's quite ugly

Reply:
In fact... maybe I've done it